docs/contracts/v2/guides/proxy-architecture/04-batch-stream.mdx

```diff
@@ -0,0 +1,297 @@
+pragma solidity >=0.8.19;
+```
+
+As mentioned in the [Overview](/contracts/v2/guides/proxy-architecture/overview), we use `Permit2` for all token
+interactions. In our example we will deal with a contract creator, and contracts can't sign transactions. Having that
+said, we will need an implementation of the [EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) standard.
+
+Let's declare a dummy contract `ERC1271` which returns the magic value (the function signature of `isValidSignature`):
+
+```solidity
+/// @notice Permit2 uses this ERC to validate contract signatures.
+contract ERC1271 {
+    function isValidSignature(bytes32, /* hash */ bytes memory /* signature */ ) public pure returns (bytes4) {
+        return this.isValidSignature.selector;
+    }
+}
+```
+
+Import the relevant symbols from `@sablier/v2-core` and `@sablier/v2-periphery`:
+
+```solidity
+import { ISablierV2LockupLinear } from "@sablier/v2-core/interfaces/ISablierV2LockupLinear.sol";
+import { LockupLinear } from "@sablier/v2-core/types/DataTypes.sol";
+import { ud60x18 } from "@sablier/v2-core/types/Math.sol";
+import { IERC20 } from "@sablier/v2-core/types/Tokens.sol";
+import { ISablierV2ProxyTarget } from "@sablier/v2-periphery/interfaces/ISablierV2ProxyTarget.sol";
+import { Batch, Broker } from "@sablier/v2-periphery/types/DataTypes.sol";
+import { IAllowanceTransfer, Permit2Params } from "@sablier/v2-periphery/types/Permit2.sol";
+import { IPRBProxy, IPRBProxyRegistry } from "@sablier/v2-periphery/types/Proxy.sol";
+```
+
+Create a contract called `BatchLockupLinearStreamCreator` that inherits `ERC1271`, and declare a constant `DAI` of type
+`IERC20`, a constant `PERMIT2` of type `IAllowanceTransfer`, a constant `PROXY_REGISTRY` of type `IPRBProxyRegistry`, an
+immutable variable `lockupLinear` of type `ISablierV2LockupLinear` and an immutable variable `proxyTarget` of type
+`ISablierV2ProxyTarget`:
+
+```solidity
+contract BatchLockupLinearStreamCreator is ERC1271 {
+    IERC20 public constant DAI = IERC20(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    IAllowanceTransfer public constant PERMIT2 = IAllowanceTransfer(0x000000000022D473030F116dDEE9F6B43aC78BA3);
+    IPRBProxyRegistry public constant PROXY_REGISTRY = IPRBProxyRegistry(0x584009E9eDe26e212182c9745F5c000191296a78);
+    ISablierV2LockupLinear public immutable lockupLinear;
+    ISablierV2ProxyTarget public immutable proxyTarget;
+}
+```
+
+In the code above, the address of the [DAI](https://makerdao.com) stablecoin, Permit2 and Proxy Registry are hard-coded.
+However, you would likely use an input parameter to allow the contract to change the assets it interacts with on a per
+transaction basis. Keep in mind that Permit2 is deployed at the same
+[address](https://docs.uniswap.org/contracts/v3/reference/deployments) on all [chains](/contracts/v2/deployments) where
+Sablier V2 is live, also Proxy Registry is deployed at the same address across these
+[chains](https://prbproxy.com/deployments).
```

Comment:
It doesn't make sense to say:

- Permit2 and ProxyRegistry are hard-coded
- Use an input parameter
- Oh, but Permit2 and ProxyRegistry are deployed at the same address